ignite/pkg/cosmosutil/gentx.go

```diff
@@ -53,44 +53,44 @@ func GentxFromPath(path string) (info GentxInfo, gentx []byte, err error) {
 	if err != nil {
 		return info, gentx, err
 	}
-	return ParseGentx(gentx)
+
+	info, err = ParseGentx(gentx)
+	return info, gentx, err
 }
 
 // ParseGentx returns GentxInfo and the gentx file in bytes
-// TODO refector. no need to return file, it's already given as gentx in the argument.
-func ParseGentx(gentx []byte) (info GentxInfo, file []byte, err error) {
-	// Try parsing Stargate gentx
-	var stargateGentx StargateGentx
-	if err := json.Unmarshal(gentx, &stargateGentx); err != nil {
-		return info, gentx, err
+func ParseGentx(gentxBz []byte) (info GentxInfo, err error) {
+	// Try parsing gentx
+	var gentx Gentx
+	if err := json.Unmarshal(gentxBz, &gentx); err != nil {
+		return info, err
```

Comment:
```suggestion
		return info, fmt.Errorf("unmarshal gentx: %w", err)
```
This helps debugging when the error occurs.